File: clients/admin/src/mutations/DeleteSeed.ts
import client from '../modules/apollo';
import mutationPromise from '../helpers/mutationPromise';
import gql from 'graphql-tag';

const mutation = gql`
  mutation DeleteSeedMutation($input: DeleteSeedInput!) {
    deleteSeed(input: $input) {
      success
      confirm
      message
    }
  }
`;

function commit(variables: DeleteSeedMutationVariables) {
  return mutationPromise((resolve, reject) => {
    client
      .mutate({
        mutation,
        variables,
        refetchQueries: ["DivisionSeedQuery"],
        awaitRefetchQueries: true
      })
      .then(({ data: { deleteSeed } }) => {
        resolve(deleteSeed as MutationResult);
      })
      .catch(error => {
        reject(error);
      });
  });
}

export default { commit };

File: app/graphql/mutations/delete_seed.rb
class Mutations::DeleteSeed < Mutations::BaseMutation
  graphql_name "DeleteSeed"

  argument :input, Inputs::DeleteSeedInput, required: true

  field :division, Types::Division, null: false
  field :success, Boolean, null: false
  field :confirm, Boolean, null: true
  field :message, String, null: false
end
